src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -412,5 +412,67 @@ public void ChainHasAssumeValidHeaderAndMarkedForDownloadWhenPresented_SecondCha
             chainedHeaderDownloadTo = connectNewHeadersResult.DownloadTo;
             chainedHeaderDownloadTo.HashBlock.Should().Be(chainBTip.HashBlock);
         }
+
+        /// <summary>
+        /// Issue 16 @ Checkpoints are enabled. After the last checkpoint, there is an assumed valid. The chain
+        /// that covers them all is presented - marked for download. After that chain that covers the last checkpoint
+        /// but doesn't cover assume valid and is longer is presented - marked for download.
+        /// </summary>
+        [Fact]
+        public void ChainHasOneCheckPointAndAssumeValid_TwoAlternativeChainsArePresented_BothChainsAreMarkedForDownload()
+        {
+            // Chain header tree setup.
+            const int initialChainSize = 4;
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            ChainedHeader initialChainTip = ctx.ExtendAChain(initialChainSize); // ie. h1=h2=h3=h4
+            cht.Initialize(initialChainTip, true);
+            ctx.ConsensusSettings.UseCheckpoints = true;
+            List<BlockHeader> listOfCurrentChainHeaders = ctx.ChainedHeaderToList(initialChainTip, initialChainSize);
+
+            // Setup a known checkpoint at header 4.
+            // Example: h1=h2=h3=(h4).
+            const int checkpointHeight = 4;
```

Comment:
In this test, we want the checkpoint to be presented by a peer, so it should not be part of the initial chain. So if our initial chain has four blocks the checkpoint should be for example at block six and assumed valid block could be at block eight.

Then we expect the first peer to present headers that include the checkpointed block six as well as the assumed valid block eight and goes further up to let's say block 10.

The second peer should present headers that include let's say the whole chain up to block seven including the checkpoint and then block eight would be different so that it does not match the assumed valid block, And it could go farther to let's say block number nine.